android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMKeyboard.java

```diff
@@ -149,9 +152,23 @@ public boolean onConsoleMessage(ConsoleMessage cm) {
           Toast.makeText(context, "Fatal Error with " + currentKeyboard +
             ". Loading default keyboard", Toast.LENGTH_LONG).show();
 
-          setKeyboard(KMManager.KMDefault_PackageID, KMManager.KMDefault_KeyboardID,
-            KMManager.KMDefault_LanguageID, KMManager.KMDefault_KeyboardName,
-            KMManager.KMDefault_LanguageName, KMManager.KMDefault_KeyboardFont, null);
+          Keyboard defaultKeyboard = KeyboardController.getInstance().getKeyboardInfo(0);
+          if (defaultKeyboard != null) {
+            setKeyboard(
+              // Revert to first keyboard in the list
+              defaultKeyboard.getPackageID(),
+              defaultKeyboard.getKeyboardID(),
+              defaultKeyboard.getLanguageID(),
+              defaultKeyboard.getKeyboardName(),
+              defaultKeyboard.getLanguageName(),
+              defaultKeyboard.getFont(),
+              defaultKeyboard.getOSKFont());
+          } else {
+            // Fallback to sil_euro_latin (though 3rd party keyboards wont have it)
+            setKeyboard(KMManager.KMDefault_PackageID, KMManager.KMDefault_KeyboardID,
+              KMManager.KMDefault_LanguageID, KMManager.KMDefault_KeyboardName,
+              KMManager.KMDefault_LanguageName, KMManager.KMDefault_KeyboardFont, null);
```

Comment:
Surely we could make a wrapper `setKeyboard` method that takes in a `Keyboard` object?  Even within just this file, we'd be using it twice and be able to drop the obnoxiously long parameter list.